Mathlib/FieldTheory/PurelyInseparable.lean

```diff
@@ -359,6 +360,23 @@ alias AlgEquiv.perfectClosure := perfectClosure.algEquivOfAlgEquiv
 
 end map
 
+/-- If `E` is a perfect field of characteristic `p`, then the (relative) perfect closure
+`perfectClosure F E` is perfect. -/
+instance perfectClosure.perfectRing (p : ℕ) [ExpChar E p]
```

Comment:
Do you assume somewhere that `p` is prime, right? This should be mentioned in the docstring (where you say "characteristic", but in the theorem you use exponential characteristic).

Reply:
I think I should say "of exponential characteristic" for accuracy. The `ExpChar` predicate implies that either `p` is one or is a prime. Also, since `E` is a field, its characteristics automatically is a prime (or zero).